buildSrc/src/main/java/org/elasticsearch/gradle/precommit/TestingConventionsTasks.java

```diff
@@ -348,7 +349,8 @@ private boolean isAnnotated(Method method, Class<?> annotation) {
         return false;
     }
 
-    private FileCollection getTestsClassPath() {
+    @Classpath
+    public FileCollection getTestsClassPath() {
```

Comment:
with compile tasks only depending on dependent projects compile tasks, the task input here needs to be explicit as it depends on jars that could not have been build before.